### PR TITLE
feat: support fallback attributions when generating codeowners file

### LINKED ISSUE #116: Feature: `attribution-fallback` config
### Suggested solution

There are cases where there may not be a codeowner that is derived off the range of commits that are being inspected. We should include a `attribution-fallback` key in the config:

```yaml
attribution:
  jpmcb:
    - john@opensauced.pizza
  brandonroberts:
    - brandon@opensauced.pizza

attribution-fallback:
  - open-sauced/engineering
```

This filed in the config could then be used to attribution files that don't have an active attribution.

**_Example_**: the codeowners generation can create valid codeowners without attributions if there are no commits in the time range for certain files:

```codeowners
/file/with/codeowner @jpmcb
/file/without/codeowner
```

With this change, it should generate:

```codeowners
/file/with/codeowner @jpmcb
/file/without/codeowner @open-sauced/engineering
```

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds the `attribution-fallback` option to the `.sauced.yaml` file
- When generating a codeowners file, if no attributions are found, the attribution fallback entries are used

TODO
- [x] Write some tests

## Related Tickets & Documents

Closes #116 

## Steps to QA

1. Run `just build`
2. Add some fallback attributions to the `.sauced.yaml` for a repository, and optionally remove all known attributions
3. Generate the codeowners file with no primary attributions
4. Note the codeowners file is generated with the fallback attributions for each file

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: cmd/generate/codeowners/output_test.go [hunk 1]
```diff
@@ -27,3 +33,37 @@ func TestCleanFilename(testRunner *testing.T) {
 		})
 	}
 }
+
+func TestGetTopContributorAttributions(testRunner *testing.T) {
+	var configSpec = config.Spec{
```
Comment (nickytonline) [inline cmd/generate/codeowners/output_test.go:38]: nit: use `:=`

```suggestion
	configSpec := config.Spec{
```

File: cmd/generate/codeowners/output_test.go [hunk 2]
```diff
@@ -27,3 +33,37 @@ func TestCleanFilename(testRunner *testing.T) {
...
+	}
+
+	results := getTopContributorAttributions(authorStats, 3, &configSpec)
+
+	assert.Equal(testRunner, len(results), 1, "Expected 1 result")
+	assert.Equal(testRunner, results[0].GitHubAlias, "brandonroberts", "Expected brandonroberts")
+}
+
+func TestGetFallbackAttributions(testRunner *testing.T) {
+	var configSpec = config.Spec{
```
Comment (nickytonline) [inline cmd/generate/codeowners/output_test.go:57]: nit: use `:=`

```suggestion
	configSpec := config.Spec{
```

**Review by nickytonline — APPROVED** [review]
Works great! 🚢

**Review by jpmcb — APPROVED** [review]
Very clean. Nice work!

**Pull request merged**